core/storage/topic/gae_models.py

```diff
@@ -21,6 +21,150 @@
 from google.appengine.ext import ndb
 
 
+class TopicSnapshotMetadataModel(base_models.BaseSnapshotMetadataModel):
+    """Storage model for the metadata for a topic snapshot."""
+    pass
+
+
+class TopicSnapshotContentModel(base_models.BaseSnapshotContentModel):
+    """Storage model for the content of a topic snapshot."""
+    pass
+
+
+class TopicModel(base_models.VersionedModel):
+    """Model for storing Topics.
+
+    This class should only be imported by the topic domain file, the
+    topic services file, and the topic model test file.
+    """
+    SNAPSHOT_METADATA_CLASS = TopicSnapshotMetadataModel
+    SNAPSHOT_CONTENT_CLASS = TopicSnapshotContentModel
+    ALLOW_REVERT = True
+
+    # The name of the topic.
+    name = ndb.StringProperty(required=True, indexed=True)
+    # This includes additional metadata like the description of the topic and
+    # some content paragraphs.
+    description = ndb.StringProperty(indexed=False)
+    # This consists of the list of canonical story ids that are part of
+    # this topic.
+    canonical_stories = ndb.StringProperty(repeated=True, indexed=False)
+    # This consists of the list of additional (non-canonical) story ids that
+    # are part of this topic.
+    additional_stories = ndb.StringProperty(repeated=True, indexed=False)
+    # This consists of the full list of skill ids that are part of this topic.
+    skills = ndb.StringProperty(repeated=True, indexed=False)
+    # The schema version for the topic.
+    schema_version = (
```

Comment:
Why is this needed? I don't see any json fields here

Reply:
Yeah, I had written this before I asked you about schema_version. Will remove this.

Reply:
Done.